src/view_mut.rs

```diff
@@ -0,0 +1,478 @@
+         OrderMapViewMut { entries: right, offset: self.offset + mid, ..self })
+    }
+
+    /// Returns the first key-value pair and a view of all the rest,
+    /// or `None` if it is empty.
+    pub fn split_first(self) -> Option<(&'a K, &'a V, Self)> {
+        if let Some((ent, rest)) = self.entries.split_first_mut() {
+            Some((&ent.key, &ent.value,
+                  OrderMapViewMut { offset: self.offset + 1, entries: rest, ..self }))
+        } else {
+            None
+        }
+    }
+
+    /// Returns the first key-value pair and a view of all the rest,
+    /// or `None` if it is empty.
+    pub fn split_first_mut(self) -> Option<(&'a mut K, &'a mut V, Self)> {
+        if let Some((ent, rest)) = self.entries.split_first_mut() {
+            Some((&mut ent.key, &mut ent.value,
+                  OrderMapViewMut { offset: self.offset + 1, entries: rest, ..self }))
+        } else {
+            None
+        }
+    }
+
+    /// Returns the last key-value pair and a view of all the rest,
+    /// or `None` if it is empty.
+    pub fn split_last(self) -> Option<(&'a K, &'a V, Self)> {
+        if let Some((ent, rest)) = self.entries.split_last_mut() {
+            Some((&ent.key, &ent.value,
+                  OrderMapViewMut { entries: rest, ..self }))
+        } else {
+            None
+        }
+    }
+
+    /// Returns the last key-value pair and a view of all the rest,
+    /// or `None` if it is empty.
+    pub fn split_last_mut(self) -> Option<(&'a mut K, &'a mut V, Self)> {
+        if let Some((ent, rest)) = self.entries.split_last_mut() {
+            Some((&mut ent.key, &mut ent.value,
+                  OrderMapViewMut { entries: rest, ..self }))
+        } else {
+            None
+        }
+    }
+}
+
+// Methods that don't use any properties (Hash / Eq) of K.
+//
+// It's cleaner to separate them out, then the compiler checks that we are not
+// using Hash + Eq at all in these methods.
+//
+// However, we should probably not let this show in the public API or docs.
+impl<'a, K, V, S> OrderMapViewMut<'a, K, V, S> {
+    /// Return probe (indices) and position (entries)
+    fn find_using<F>(&self, hash: HashValue, key_eq: F) -> Option<(usize, usize)>
+        where F: Fn(&Bucket<K, V>) -> bool,
+    {
+        dispatch_32_vs_64!(self.find_using_impl::<_>(hash, key_eq))
+    }
+
+    fn find_using_impl<Sz, F>(&self, hash: HashValue, key_eq: F) -> Option<(usize, usize)>
+        where F: Fn(&Bucket<K, V>) -> bool,
+              Sz: Size,
+    {
+        debug_assert!(self.len() > 0);
+        let mut probe = desired_pos(self.mask, hash);
+        let mut dist = 0;
+        probe_loop!(probe < self.indices.len(), {
+            if let Some((i, hash_proxy)) = self.indices[probe].resolve::<Sz>() {
+                // Unfortunately, we can only look at indexes within our slice
+                // of entries, which might mean we'll keep probing farther than
+                // it would be strictly necessary.
```

Comment:
We can remedy this (often); by asking for the short hash that is stored in the indices; it is accessible when usize is 64-bit and the entry capacity fits in 32 bits. It requires breaking down `get_short_hash` and using something like this:

```rust
/// Get the hash from the indices if possible.
fn try_short_hash<K, V>(&self, index: usize) -> Option<ShortHash<Sz>> {
    if Sz::is_64_bit() {
        None
    } else {
        Some(ShortHash(self.0, PhantomData))
    }
}
```